plugins/techdocs-backend/src/cache/TechDocsCache.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { CacheClient } from '@backstage/backend-common';
+import { assertError } from '@backstage/errors';
+import { Config } from '@backstage/config';
+import { Logger } from 'winston';
+
+export class CacheInvalidationError extends Error {
```

Comment:
[CustomErrorBase](https://github.com/backstage/backstage/blob/60020e8ae4a4b9814ec173042e1ea504b3c72d9e/packages/errors/src/errors/CustomErrorBase.ts#L35) from the errors package is a good helper for stuff like this.